olad/PluginManager.cpp

```diff
@@ -73,15 +72,14 @@ void PluginManager::LoadAll() {
         OLA_INFO << "Skipping " << plugin->Name() << " because it was disabled";
         continue;
       }
-      enabled_plugins.push_back(plugin);
-      enabled_plugin_ids.insert(plugin->Id());
+      STLInsertIfNotPresent(&m_enabled_plugins, plugin->Id(), plugin);
     }
   }
 
   // The second pass checks for conflicts and starts each plugin
-  vector<AbstractPlugin*>::iterator plugin_iter = enabled_plugins.begin();
-  for (; plugin_iter != enabled_plugins.end(); ++plugin_iter) {
-    AbstractPlugin *plugin = *plugin_iter;
+  PluginMap::iterator plugin_iter = m_enabled_plugins.begin();
+  for (; plugin_iter != m_enabled_plugins.end(); ++plugin_iter) {
+    AbstractPlugin *plugin = &(*plugin_iter->second);
```

Comment:
I don't know if you want AbstractPlugin* changed to a straight AbstractPlugin and the necessary changes in the rest of this code?